### PR TITLE
kgo: add horizontal dataplane autoscaling section

### PR DESCRIPTION
**Pull request by pmalek**
### Description

Add KGO section for DataPlane horizontal autoscaling.

Related work: https://github.com/Kong/gateway-operator/pull/1281

This is targeting `kgo-1.2.x` because the changes from this PR are relevant for KGO 1.2.x which is yet unreleased.

### Testing instructions

Preview link: https://deploy-preview-6710--kongdocs.netlify.app/gateway-operator/latest/production/horizontal-autoscaling/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Comment by pmalek** [comment]
Ready for review but please do not merge until https://github.com/Kong/gateway-operator/issues/1217 gets merged.

File: app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md
Comment (pmalek) [inline app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md:8]: Not sure if this is necessary given that this page will only be rendered in 1.2.x and up due to not being included in older nav files.

Technically ( I can be proven wrong ) this is not necessary given the above so I'll defer to @Kong/docs-maintainers if this makes sense.

File: app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md
Comment (pmalek) [inline app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md:88]: I wanted to add a note linking to https://deploy-preview-6710--kongdocs.netlify.app/gateway-operator/1.2.x/reference/custom-resources/#dataplanedeploymentoptions but for some reason this doesn't include the `scaling` section which I have definitely added https://github.com/Kong/docs.konghq.com/pull/6710/files#diff-4d19265cfc4982d86c9005fa2bdd422a19040d4874080edaae1c617df1d9091dR243.

I suppose that's because docs still think that 1.1 is the latest version even though I have added 1.2 in `app/_data/kong_versions.yml`.

Any ideas how to make the new CRD page have the changed (new) content?

Comment (pmalek) [reply app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md:88]: cc @Kong/docs-maintainers

File: app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md
Comment (mheap) [reply app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md:8]: This is correct. You can remove the `if_version` as it's only included in 1.2.x+ nav files

Comment (pmalek) [reply app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md]: Gotcha, remove in this case. I was worried that users could still tinker with the version in the URL and get the site for earlier versions.

**Comment by pmalek** [comment]
https://github.com/Kong/gateway-operator/issues/1217 got merged and links have been updated. 

Not sure why the broken links CI job is failing when the offending link was already removed 🤔

File: app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md
Comment (pmalek) [reply app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md]: This is solved now, thanks to @mheap 🙇

**Comment by mheap** [comment]
The answer to all of the above is that `kgo-1.2.x` wasn't in the list of deployed branches. Broken links run against preview apps. The preview wasn't being updated as Netlify was not deploying additional commits due to base branch change

File: app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md
Comment (mflendrich) [inline app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md:1]: Is there a good reason for autoscaling to live under the "Konnect Quick Start"?

Thoughts from my side:
1. HPA doesn't belong in the "quick" zero-to-one path
2. This is now a feature for the Konnect mode only, but this reality will almost certainly change in a future version
3. There are concepts in autoscaling that go beyond a "quick start" for somebody running Konnect :
    1. In this doc: the load test
    2. In general / in a broader sense: tweaks, best practices, interactions with other mechanisms (such as B/G rollouts), explanation of what modes support HPA and which don't

Suggestion: leave the quickstart "quick" and put the HPA documentation in the "production use" cookbook with targeted recipes for particular features - under "Production Deployment".

Comment (pmalek) [reply app/_src/gateway-operator/get-started/konnect/horizontal-autoscaling.md:1]: 👍 Re: moving this outside of quickstart, makes sense as that's not really a topic one would like to read when trying to figure out how to start using KGO.

File: app/_data/docs_nav_kgo_1.2.x.yml
Comment (mflendrich) [inline app/_data/docs_nav_kgo_1.2.x.yml:76]:
```suggestion
      - text: Autoscaling
```
or
```suggestion
      - text: Autoscaling Data Planes
```

- this way it won't wrap around the line break (on my firefox-on-mac display it does)
- later we can make "autoscaling" a group with more specialized (horizontal / vertical / by cpu, latency, etc) entries underneath

**Review by mheap — APPROVED** [review]
@pmalek I gave the PR a copyedit + added a bit of context around autoscaling values. Please merge as soon as you're happy

**Comment by pmalek** [comment]
> @pmalek I gave the PR a copyedit + added a bit of context around autoscaling values. Please merge as soon as you're happy

Thanks! Nice improvement over what was proposed. One thing I have not mentioned in the doc but I'm considering now is mentioning the operator feature which was implemented as part of the horizontal autoscaling, which sets the initial `Deployment` replicas to `spec.deployment.scaling.horizontal.minReplicas`. This cuts down on time that is needed (by the HPA) to scale out to `minReplicas`. @mheap Do you believe that would be worth adding in the doc?

**Comment by pmalek** [comment]
Merging now, I can address https://github.com/Kong/docs.konghq.com/pull/6710#issuecomment-1883468243 in a separate PR. Thanks everyone for input!

**Pull request merged**
